tests/test_cosine_greedy_numba.py

```diff
@@ -0,0 +1,80 @@
+import numpy
+import pytest
+from matchms import Spectrum
+from matchms.filtering import normalize_intensities
+from matchms.similarity import CosineGreedyNumba
+
+
+def test_cosine_greedy_without_parameters():
+
+    spectrum_1 = Spectrum(mz=numpy.array([100, 150, 200, 300, 500, 510, 1100], dtype="float"),
+                          intensities=numpy.array([700, 200, 100, 1000, 200, 5, 500], dtype="float"))
+
+    spectrum_2 = Spectrum(mz=numpy.array([100, 140, 190, 300, 490, 510, 1090], dtype="float"),
+                          intensities=numpy.array([700, 200, 100, 1000, 200, 5, 500], dtype="float"))
+
+    norm_spectrum_1 = normalize_intensities(spectrum_1)
+    norm_spectrum_2 = normalize_intensities(spectrum_2)
+    cosine_greedy_numba = CosineGreedyNumba()
+    score, n_matches = cosine_greedy_numba(norm_spectrum_1, norm_spectrum_2)
+
+    assert score == pytest.approx(0.81421, 0.0001), "Expected different cosine score."
```

Comment:
Could you again say something about the 'magic' number that is the expected score? Where did it come from and how to reproduce?

Reply:
I replaced most test cases with simpler cases.  Expected scores are now calculated in the test instead of just magically stated. I didn't go as far as to explain every step of the score calculation but in principle it is now possible to understand where they come from (I hope).

Reply:
👍